sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -841,13 +841,31 @@ def test_csr_polynomial_expansion_index_overflow_deg2(
     pf = PolynomialFeatures(
         interaction_only=interaction_only, include_bias=include_bias, degree=2
     )
-    X_trans = pf.fit_transform(X)
 
-    expected_dtype = (
-        np.int64 if X_trans.indices.max() > np.iinfo(np.int32).max else np.int32
+    # `scipy.sparse.hstack` breaks in scipy<1.9.2 when
+    # `n_output_features_ > max_int32`. Test that we provide
+    # an informative error message.
+    if (
+        sp_version < parse_version("1.9.2")
+        and n_features == 65535
+        and not interaction_only
+    ):
```

Comment:
```suggestion
    if (
        sp_version < parse_version("1.9.2")
        and n_features >= 65535
        and not interaction_only
    ):
```

Reply:
Actually the condition is a bit more specific, since if `n_features` is sufficiently large then this error won't be raised (component matrices will have `idx_dtype=int64`). Since the case is specifically designed to be `65535` in the test I'd rather keep it as-is. I think it is fine, since the causes are documented in the error and this test value specifically guarantees it. What do you think?

Reply:
This works for me. I think we can include an inline comment based on your remark:

```suggestion
    if (
        sp_version < parse_version("1.9.2")
        # When `n_features>65535`, `scipy.sparse.hstack` use the appropriate dtype
        # for representing indices and indptr.
        # `n_features==65535` is a corner case that we need to test.
        and n_features == 65535
        and not interaction_only
    ):
```

What do you think?